domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java

```diff
@@ -0,0 +1,122 @@
+package org.egov.ifixespipeline.consumers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.request.RequestHeader;
+import org.egov.ifixespipeline.models.*;
+import org.egov.ifixespipeline.producer.Producer;
+import org.egov.ifixespipeline.repository.ServiceRequestRepository;
+import org.egov.ifixespipeline.service.FiscalDataEnrichmentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@Service
+@Slf4j
+public class IfixElasticSearchPipelineListener {
+
+    @Autowired
+    private FiscalDataEnrichmentService fiscalDataEnrichmentService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ServiceRequestRepository serviceRequestRepository;
+
+    @Autowired
+    private Producer producer;
+
+    @Value("${fiscal.event.es.push.topic}")
+    private String indexFiscalEventsTopic;
+
+    @Value("${fiscal.event.kafka.push.topic}")
+    private String fiscalEventsNewRecordsTopic;
+
+    @Value("${ifix.master.data.service.host}")
+    private String ifixMasterDataServiceHost;
+
+    @Value("${ifix.master.data.service.search.endpoint}")
+    private String ifixMasterDataServiceSearchEndpoint;
+
+    @Value("${coa.electricity.head.name}")
+    private String electricityCoaHeadName;
+
+    @Value("${coa.operations.head.name}")
+    private String operationsCoaHeadName;
+
+    @Value("${coa.salary.head.name}")
+    private String salaryCoaHeadName;
+
+    private Map<String, HashSet<String>> expenditureTypeVsUuidsMap;
+
+    /**
+     * Kafka consumer
+     *
+     * @param record
+     * @param topic
+     */
+    @KafkaListener(topics = { "${fiscal.event.kafka.push.topic}", "${fiscal.event.migration.origin.push.topic}"})
+    public void listen(HashMap<String, Object> record, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        try {
+            FiscalEventRequest incomingData = objectMapper.convertValue(record, FiscalEventRequest.class);
+
+            // Enrich hierarchy map in only new records. In case of migration records, skip enrichment as they are already getting enriched in migration toolkit.
+            if(topic.equalsIgnoreCase(fiscalEventsNewRecordsTopic)) {
+                expenditureTypeVsUuidsMap = loadExpenditureTypeVsUuidMap(incomingData.getFiscalEvent().getTenantId());
```

Comment:
Load the map during initiation and check if the key is present for the incoming tenant. If new tenant is added, in that case key won't be present and the map should be updated by adding the new tenant entry